warehouse/query-core/src/main/java/datawave/core/iterators/DatawaveFieldIndexRangeIteratorJexl.java

```diff
@@ -109,60 +117,86 @@ public String toString() {
     
     @Override
     protected List<Range> buildBoundingFiRanges(Text rowId, Text fiName, Text fieldValue) {
```

Comment:
please add unit tests on the boundary conditions of this method

Reply:
Got it.  Check out the latest, please.